toxencryptsave/constants.hpp

```diff
@@ -0,0 +1,31 @@
+#ifndef TOXENCRYPTSAVE_CONSTANTS_H
+#define TOXENCRYPTSAVE_CONSTANTS_H
+
+#include <stddef.h>
+#include <stdint.h>
+
+// hardcoded constants
+// if they change, userdata will be lost
+
+#define TOXENCRYPTSAVE_MAGIC_NUMBER "toxEsave"
+#define TOXENCRYPTSAVE_MAGIC_LENGTH 8U
+
+#define TOXENCRYPTSAVE_SALT_LENGTH 32U
+#define TOXENCRYPTSAVE_KEY_LENGTH 32U
+#define TOXENCRYPTSAVE_NONCEBYTES 24U
+#define TOXENCRYPTSAVE_ENCRYPTION_EXTRA_LENGTH 80U
+
+namespace toxencryptsave
+{
+    namespace constants
+    {
+        extern const char* magic;
+        static const size_t magic_len = TOXENCRYPTSAVE_MAGIC_LENGTH;
```

Comment:
static variables in headers probably is not a good idea. Use extern static. Or can we use constexpr?
And why do we need both, defines and constants?
May be will be better
enum SizeConstants : size_t { ...}
?